app/javascript/mastodon/features/ui/index.js

```diff
@@ -123,13 +127,28 @@ export default class UI extends React.PureComponent {
     this.setState({ draggingOver: false });
   }
 
+  handleServiceWorkerPostMessage = ({ data }) => {
+    if (data.type === 'navigate') {
+      const url = new URL(data.url);
+
+      if (url.host === location.host && url.pathname.startsWith('/web/')) {
+        this.context.router.history.push(url.pathname.slice('/web/'.length - 1));
```

Comment:
should we ever postMessage a request to go somewhere that's not the local domain? This feels like a security hole slash unnecessary complexity.